GeneralsMD/Code/GameEngine/Source/GameClient/GUI/GUICallbacks/Menus/OptionsMenu.cpp

```diff
@@ -1630,22 +1629,21 @@ void OptionsMenuInit( WindowLayout *layout, void *userData )
 		TheDisplay->getDisplayModeDescription(i,&xres,&yres,&bitDepth);
 		str.format(L"%d x %d",xres,yres);
 		GadgetComboBoxAddEntry( comboBoxResolution, str, color);
-		if (xres == 800 && yres == 600)	//keep track of default mode in case we need it.
-			defaultResIndex=i;
 		if (xres == selectedXRes && yres == selectedYRes)
 			selectedResIndex=i;
 	}
 
 	if (selectedResIndex == -1)	//check if saved mode no longer available
-	{	//pick default resolution
-		selectedXRes = 800;
-		selectedXRes = 600;
-		selectedResIndex = defaultResIndex;
+	{
+		// TheSuperHackers @tweak xezon 08/06/2025 Now adds the current resolution instead of defaulting to 800 x 600.
```

Comment:
I think this needs testing in fullscreen. It might only work in Windowed Mode. Fullscreen will offer less freedoms in choosing custom resolutions.

Reply:
Works in fullscreen.